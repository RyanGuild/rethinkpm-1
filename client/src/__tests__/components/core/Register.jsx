import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import Register from '../../../components/core/Register';
import render from '../../../__utils__/renderHelper';
import {useLocation} from 'react-router-dom';

test('renders and snapshots', () => {
  let setAlert = jest.fn();
  let register = jest.fn();
  let isAuthenticated = false;
  let location;
  let {container} = render(() => {
    location = useLocation();
    return (
      <Register
        setAlert={setAlert}
        register={register}
        isAuthenticated={isAuthenticated}
      />
    );
  });
  expect(container).not.toBeNull();
  expect(location.pathname).toEqual('/');
  expect(container).toMatchSnapshot();
});
