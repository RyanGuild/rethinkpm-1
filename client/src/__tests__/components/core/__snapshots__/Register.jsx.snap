// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<div>
  <h1
    class="large text-primary"
  >
    Sign Up
  </h1>
  <p
    class="lead"
  >
    <i
      class="fas fa-user"
    />
     Create Your Account
  </p>
  <form
    class="form"
  >
    <div
      class="form-group"
    >
      <input
        name="name"
        placeholder="Name"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <input
        name="email"
        placeholder="Email Address"
        type="email"
        value=""
      />
      <small
        class="form-text"
      >
        This site uses Gravatar so if you want a profile image, use a Gravatar email
      </small>
    </div>
    <div
      class="form-group"
    >
      <input
        name="phone"
        placeholder="Mobile Phone"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <input
        name="password"
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <input
        name="password2"
        placeholder="Confirm Password"
        type="password"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <small
        class="form-text"
      >
        Choose the option(s) that best descibe you
      </small>
      <input
        name="owner"
        type="checkbox"
        value="false"
      />
       
      I am a Property Owner Looking for Property Managment Solutions
      <br />
      <input
        name="investor"
        type="checkbox"
        value="false"
      />
       
      I am Real Estate Investor Looking To Purchase My Next Deal 
      <br />
      <input
        name="vendor"
        type="checkbox"
        value="false"
      />
       
      I am Contractor Interested In Joining A network Of Service Pro's
       
      <br />
    </div>
    <input
      class="btn btn-primary"
      type="submit"
      value="Register"
    />
  </form>
  <p
    class="my-1"
  >
    Already have an account? 
    <a
      href="/login"
    >
      Sign In
    </a>
  </p>
</div>
`;
